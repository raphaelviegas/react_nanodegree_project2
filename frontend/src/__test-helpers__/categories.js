export const initialState = { categories: [] }

export const categories = [
    {
        name: 'react',
        path: 'react'
    },
    {
        name: 'redux',
        path: 'redux'
    },
    {
        name: 'udacity',
        path: 'udacity'
    }
]