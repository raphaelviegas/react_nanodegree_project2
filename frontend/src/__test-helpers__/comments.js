const uuidv4 = require('uuid/v4')

export const initialState = { comments: [] }

export const comments = [
    {
        id: '894tuq4ut84ut8v4t8wun89g',
        parentId: "8xf0y6ziyjabvozdd253nd",
        timestamp: 1468166872634,
        body: 'Hi there! I am a COMMENT.',
        author: 'thingtwo',
        voteScore: 6,
        deleted: false,
        parentDeleted: false
    },
    {
        id: '8tu4bsun805n8un48ve89',
        parentId: "8xf0y6ziyjabvozdd253nd",
        timestamp: 1469479767190,
        body: 'Comments. Are. Cool.',
        author: 'thingone',
        voteScore: -5,
        deleted: false,
        parentDeleted: false
    }
]

export const getCommentsId = '894tuq4ut84ut8v4t8wun89g'

export const removeComment = '894tuq4ut84ut8v4t8wun89g'

export const voteComment = {
    id: '894tuq4ut84ut8v4t8wun89g',
    voteScore: 10
}

let id = uuidv4()
export const addCommentInfo = {
    parentId: '8xf0y6ziyjabvozdd253nd',
    author: 'Test Author',
    body: 'Test Body',
    id
}

export const editCommentInfo = {
    id: '894tuq4ut84ut8v4t8wun89g',
    editContent: {
        body: 'Edited comment body',
        timestamp: Date.now()
    }
}