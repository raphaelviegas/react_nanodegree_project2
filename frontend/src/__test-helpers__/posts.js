const uuidv4 = require('uuid/v4')

export const initialState = {
    posts: [],
    sortByOption: '-voteScore',
    invalidPost: false,
    loadingPosts: true,
}

export const posts = [
    {
        id: '8xf0y6ziyjabvozdd253nd',
        timestamp: 1467166872634,
        title: 'Udacity is the best place to learn React',
        body: 'Everyone says so after all.',
        author: 'thingtwo',
        category: 'react',
        voteScore: 10,
        deleted: false,
        commentCount: 2
    },
    {
        id: '6ni6ok3ym7mf1p33lnez',
        timestamp: 1468479767190,
        title: 'Learn Redux in 10 minutes!',
        body: 'Just kidding. It takes more than 10 minutes to learn technology.',
        author: 'thingone',
        category: 'redux',
        voteScore: -5,
        deleted: false,
        commentCount: 0
    }
]

export const removePost = '8xf0y6ziyjabvozdd253nd'

export const votePost = {
    id: '8xf0y6ziyjabvozdd253nd',
    voteScore: 10
}

let id = uuidv4()
export const addPostInfo = {
    author: "Test Author",
    body: "Test Body",
    category: "Test",
    title: "Test Title",
    timestamp: Date.now(),
    id
}

export const editPostInfo = {
    id: '8xf0y6ziyjabvozdd253nd',
    editContent: {
        title: 'Edited post title',
        body: 'Edited post body'
    }
}