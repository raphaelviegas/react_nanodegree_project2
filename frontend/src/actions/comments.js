import * as ServerAPI from '../helpers/ServerAPI'

export const LOAD_COMMENTS = 'LOAD_COMMENTS'
export const REMOVE_COMMENT = 'REMOVE_COMMENT'
export const VOTE_COMMENT = 'VOTE_COMMENT'
export const ADD_COMMENT = 'ADD_COMMENT'
export const EDIT_COMMENT = 'EDIT_COMMENT'

//Local state
export function loadComments (comments) {
  return {
    type: LOAD_COMMENTS,
    payload: comments
  }
}
export function voteComment (id, voteScore) {
  return {
    type: VOTE_COMMENT,
    payload: {id, voteScore }
  }
}

export function removeComment (id) {
  return {
    type: REMOVE_COMMENT,
    payload: id
  }
}

export function addComment(comment) {
  return {
    type: ADD_COMMENT,
    payload: comment
  }
}

export function editComment (id, editContent) {
  return {
    type: EDIT_COMMENT,
    payload: {
      id,
      editContent
    }
  }
}

//Fetch requests
export const fetchComments = (postId) => dispatch => (
  ServerAPI.getComments(postId)
    .then(comments => dispatch(loadComments(comments)))
)

export const fetchVoteComment = (id, option) => dispatch => (
  ServerAPI.vote('comments', id, option)
    .then(comment => dispatch(voteComment(comment.id, comment.voteScore)))
)

export const fetchRemoveComment = (id) => dispatch => (
  ServerAPI.remove('comments', id)
    .then(comment => dispatch(removeComment(comment.id)))
)

export const fetchAddComment = (commentInfo) => dispatch => (
  ServerAPI.add('comments', commentInfo)
    .then(comment => dispatch(addComment(comment)))
)

export const fetchEditComment = (id, editContent) => dispatch => (
  ServerAPI.edit('comments', id, editContent)
    .then(()=> dispatch(editComment(id, editContent)))
)
