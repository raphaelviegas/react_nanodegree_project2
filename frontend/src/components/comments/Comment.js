import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Moment from 'react-moment';
import VoteScore from '../shared/VoteScore'

class Comment extends Component {
    static propTypes = {
        commentInfo: PropTypes.object.isRequired,
        onRemoveComment: PropTypes.func.isRequired,
        onEditComment: PropTypes.func.isRequired,
    }

    removeComment(id) {
        this.props.onRemoveComment(id)
    }

    editComment(id, body) {
        this.props.onEditComment(id, body)
    }

    render() {
        const { author, timestamp, body, voteScore, id } = this.props.commentInfo
        return (
            <li>
                <div className="card">
                    <div className="card-content">
                        <span className="card-title comment-title">Author: {author}</span>
                        <span className="card-date comment-date"><Moment format="DD/MMM/YY">{timestamp}</Moment></span>
                        <p className="card-body comment-body">{body}</p>
                    </div>
                    <div className="card-action comment-action">
                        <span className="comment-action-edit" onClick={() => this.editComment(id, body)}>Edit</span>
                        <span className="comment-action-remove" onClick={() => this.removeComment(id)}>Delete</span>
                    </div>
                    <div className="comment-vote-score">
                        <VoteScore size={12} score={voteScore} type='comment' id={id} />
                    </div>
                </div>
            </li>
        )
    }
}

export default Comment
