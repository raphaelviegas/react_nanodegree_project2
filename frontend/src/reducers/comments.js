import {
    LOAD_COMMENTS, 
    REMOVE_COMMENT, 
    VOTE_COMMENT,
    EDIT_COMMENT,
    ADD_COMMENT
} from '../actions/comments'

function commentsReducer (state = {
    comments: [],
}, action) {
    switch (action.type) {
        case LOAD_COMMENTS:
            state = {
                ...state,
                comments: action.payload
            }
            return state
        case REMOVE_COMMENT:
            state ={
                ...state,
                comments: state.comments.filter(comment => comment.id !== action.payload)
            }
            return state
        case VOTE_COMMENT:
            state ={
                ...state,
                comments: state.comments.map(comment => {
                    if (comment.id === action.payload.id) {
                        comment.voteScore = action.payload.voteScore
                    }
                    return comment
                })
            }
            return state
        case ADD_COMMENT:
            let comments = state.comments.concat([action.payload])
            state = {
                ...state,
                comments
            }
            return state
        case EDIT_COMMENT:
            state = {
                ...state,
                comments: state.comments.map(comment => {
                    if (comment.id === action.payload.id) {
                        comment.timestamp = action.payload.editContent.timestamp
                        comment.body = action.payload.editContent.body
                    }
                    return comment
                })
            }
            return state
        default:
            return state
    }
}

export default commentsReducer