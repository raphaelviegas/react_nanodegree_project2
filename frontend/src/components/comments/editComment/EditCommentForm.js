import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import {fetchEditComment} from '../../../actions/comments'

export class EditCommentForm extends Component {
    static propTypes = {
        commentId: PropTypes.string.isRequired,
        commentBody: PropTypes.string.isRequired,
        closeModal: PropTypes.func.isRequired
    }

    state = {
        body: '',
    }

    fillDefaultValue (body) {
        this.setState({
            body
        })
    }

    componentDidMount () {
        this.fillDefaultValue (this.props.commentBody)
    }

    componentDidUpdate(prevProps){
        if (this.props.commentBody !== prevProps.commentBody) {
            this.fillDefaultValue (this.props.commentBody)
        }
    }

    onFormChange = (value, inputField) => {
        this.setState({[inputField]: value})
    }

    onFormSubmit = (body, id) => {
        let timestamp = Date.now()
        let commentInfo = {body, timestamp}
        this.props.fetchEditComment(id, commentInfo)
        this.props.closeModal()
    }

    render() {
        const {body} = this.state
        const {commentId} = this.props
        return (
            <div>
                <div className="modal-content">
                    <h4>Edit comment</h4>
                    <form className="col s12">
                        <div className="row form-row">
                            <div className="input-field col s12">
                                <textarea 
                                    placeholder="Type some text" 
                                    type="text" 
                                    value={body}
                                    onChange={(event) => this.onFormChange(event.target.value, 'body')}
                                    className="materialize-textarea" />
                                <label>Comment Body</label>
                            </div>
                        </div>
                    </form>
                </div>
                <div className="modal-footer">
                    <button 
                        className="btn white-text cyan darken-4 btn-flat"
                        onClick={() => this.onFormSubmit(body, commentId)}
                    >Edit Comment</button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        commentId: ownProps.commentId,
        commentBody: ownProps.commentBody,
        closeModal: ownProps.closeModal
    }
}
  
const mapDispatchToProps = (dispatch) => {
    return {
        fetchEditComment: (id, commentInfo) => dispatch(fetchEditComment(id, commentInfo))
    }
}
  
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(EditCommentForm));
