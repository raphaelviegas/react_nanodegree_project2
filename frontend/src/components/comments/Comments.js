import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { fetchComments, fetchRemoveComment } from '../../actions/comments'
import MaterialIcon from 'material-icons-react'
import sortBy from 'sort-by'
import AddCommentButton from './addComment/AddCommentButton'
import EditCommentModal from './editComment/EditCommentModal'
import Comment from './Comment'
import M from "materialize-css/dist/js/materialize.min.js"

export class Comments extends Component {
    static propTypes = {
        parentId: PropTypes.string.isRequired,
        comments: PropTypes.array.isRequired
    }

    state = {
        sortByOption: '-voteScore',
        commentId: '',
        commentBody: '',
        editModalOpen: ''
    }

    showEditModal = (id, body) => this.setState(
        () => ({ editModalOpen: 'modal-open', commentId: id, commentBody: body })
    )

    hideEditModal = () => this.setState(
        () => ({ editModalOpen: '', commentId: '', commentBody: '', })
    )

    componentDidMount() {
        this.props.fetchComments(this.props.parentId)
        let elems = document.querySelectorAll('.dropdown-trigger');
        // eslint-disable-next-line
        let instances = M.Dropdown.init(elems, { alignment: 'right', constrainWidth: false, hover: false });
    }

    removeComment = (id) => {
        this.props.fetchRemoveComment(id)
    }

    changeSortCriteria = (option) => {
        switch (option) {
            case 'voteScore':
                this.setState({ sortByOption: 'voteScore' })
                break
            case '-voteScore':
                this.setState({ sortByOption: '-voteScore' })
                break
            case 'timestamp':
                this.setState({ sortByOption: 'timestamp' })
                break
            case '-timestamp':
                this.setState({ sortByOption: '-timestamp' })
                break
            default:
                break
        }
    }

    render() {
        const { comments, parentId } = this.props
        const commentCount = comments.length
        const { sortByOption, commentId, commentBody, editModalOpen } = this.state
        //Sort By
        comments.sort(sortBy(sortByOption))
        return (
            <div className="post-comments">
                <div className="post-comments-title">
                    Comments: {commentCount}
                    <div className="order-by">
                        <a className='dropdown-trigger' data-target='dropdown1'><MaterialIcon icon="sort" size={22} /></a>
                        <ul id='dropdown1' className='dropdown-content'>
                            <li><span>Vote Score</span></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('voteScore')}>Ascending</a></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('-voteScore')}>Descending</a></li>
                            <li className="divider" tabIndex="-1"></li>
                            <li><span>Date</span></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('timestamp')}>Ascending</a></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('-timestamp')}>Descending</a></li>
                        </ul>
                    </div>
                </div>
                <div className="divider"></div>
                {commentCount === 0
                    ? <div className="noComments"><h6>No comments to show</h6></div>
                    : <ul className="comments">
                        {comments.map(comment => (
                            <Comment
                                commentInfo={comment}
                                key={comment.id}
                                onRemoveComment={this.removeComment}
                                onEditComment={this.showEditModal}
                            />
                        ))}
                    </ul>
                }
                <AddCommentButton parentId={parentId} />
                <EditCommentModal
                    modalOpen={editModalOpen}
                    commentId={commentId}
                    commentBody={commentBody}
                    onCloseModal={this.hideEditModal}
                />
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        parentId: ownProps.parentId,
        comments: state.commentsReducer.comments
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchComments: (parentId) => dispatch(fetchComments(parentId)),
        fetchRemoveComment: (id) => dispatch(fetchRemoveComment(id))
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Comments));