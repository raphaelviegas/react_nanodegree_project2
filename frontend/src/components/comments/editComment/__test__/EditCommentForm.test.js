import React from 'react'
import { shallow } from 'enzyme'
import { EditCommentForm } from '../EditCommentForm'
import { editCommentInfo } from '../../../../__test-helpers__/comments'

describe('<EditCommentForm />', () => {
    const setup = {
        commentId: editCommentInfo.id,
        commentBody: editCommentInfo.editContent.body,
        fetchEditComment: jest.fn(),
        closeModal: jest.fn()
    }
    let wrapper
    beforeEach(() => {
        setup.fetchEditComment.mockClear()
        setup.closeModal.mockClear()
        wrapper = shallow(<EditCommentForm {...setup} />)
    })

    it('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Should have initial state with empty values', () => {
        const wrapperNoLifecicle = shallow(<EditCommentForm {...setup} />, { disableLifecycleMethods: true })
        expect(wrapperNoLifecicle.state()).toEqual({ body: '' })
    })

    it('Should expect method fillDefaultValue to update the state.body', () => {
        const wrapperNoLifecicle = shallow(<EditCommentForm {...setup} />, { disableLifecycleMethods: true })
        expect(wrapperNoLifecicle.state()).toEqual({ body: '' })
        wrapperNoLifecicle.instance().fillDefaultValue('New body value')
        expect(wrapperNoLifecicle.state()).toEqual({ body: 'New body value' })
    })

    it('Should update state with values passed as props', () => {
        expect(wrapper.state()).toEqual({ body: setup.commentBody })
    })

    it('Should update the body value when props are updated', () => {
        const mockfillDefaultValue = jest.fn()
        const newWrapper = shallow(<EditCommentForm {...setup} />)
        newWrapper.instance().fillDefaultValue = mockfillDefaultValue
        expect(mockfillDefaultValue).toHaveBeenCalledTimes(0)
        newWrapper.setProps({ commentBody: 'some comment body' })
        expect(mockfillDefaultValue).toHaveBeenCalledTimes(1)
    })

    it('Should change the state.body after onChange event on title input', () => {
        expect(wrapper.state()).toEqual({ body: setup.commentBody })
        wrapper.find('textarea').simulate('change', { target: { value: 'New Body' } })
        expect(wrapper.state()).toEqual({ body: 'New Body' })
    })

    it('Should call fetchEditComment and closeModal on form submit', () => {
        expect(setup.fetchEditComment).toHaveBeenCalledTimes(0)
        expect(setup.closeModal).toHaveBeenCalledTimes(0)
        wrapper.find('button').simulate('click')
        expect(setup.fetchEditComment).toHaveBeenCalledTimes(1)
        expect(setup.closeModal).toHaveBeenCalledTimes(1)
    })
})