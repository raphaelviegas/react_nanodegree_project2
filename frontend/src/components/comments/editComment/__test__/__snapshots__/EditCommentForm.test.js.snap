// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditCommentForm /> Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditCommentForm
    closeModal={[Function]}
    commentBody="Edited comment body"
    commentId="894tuq4ut84ut8v4t8wun89g"
    fetchEditComment={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="modal-content"
>
          <h4>
                    Edit comment
          </h4>
          <form
                    className="col s12"
          >
                    <div
                              className="row form-row"
                    >
                              <div
                                        className="input-field col s12"
                              >
                                        <textarea
                                                  className="materialize-textarea"
                                                  onChange={[Function]}
                                                  placeholder="Type some text"
                                                  type="text"
                                                  value="Edited comment body"
                                        />
                                        <label>
                                                  Comment Body
                                        </label>
                              </div>
                    </div>
          </form>
</div>,
        <div
          className="modal-footer"
>
          <button
                    className="btn white-text cyan darken-4 btn-flat"
                    onClick={[Function]}
          >
                    Edit Comment
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Edit comment
</h4>,
            <form
              className="col s12"
>
              <div
                            className="row form-row"
              >
                            <div
                                          className="input-field col s12"
                            >
                                          <textarea
                                                        className="materialize-textarea"
                                                        onChange={[Function]}
                                                        placeholder="Type some text"
                                                        type="text"
                                                        value="Edited comment body"
                                          />
                                          <label>
                                                        Comment Body
                                          </label>
                            </div>
              </div>
</form>,
          ],
          "className": "modal-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit comment",
            },
            "ref": null,
            "rendered": "Edit comment",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row form-row"
>
                <div
                                className="input-field col s12"
                >
                                <textarea
                                                className="materialize-textarea"
                                                onChange={[Function]}
                                                placeholder="Type some text"
                                                type="text"
                                                value="Edited comment body"
                                />
                                <label>
                                                Comment Body
                                </label>
                </div>
</div>,
              "className": "col s12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="input-field col s12"
>
                  <textarea
                                    className="materialize-textarea"
                                    onChange={[Function]}
                                    placeholder="Type some text"
                                    type="text"
                                    value="Edited comment body"
                  />
                  <label>
                                    Comment Body
                  </label>
</div>,
                "className": "row form-row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <textarea
                      className="materialize-textarea"
                      onChange={[Function]}
                      placeholder="Type some text"
                      type="text"
                      value="Edited comment body"
/>,
                    <label>
                      Comment Body
</label>,
                  ],
                  "className": "input-field col s12",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "materialize-textarea",
                      "onChange": [Function],
                      "placeholder": "Type some text",
                      "type": "text",
                      "value": "Edited comment body",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "textarea",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Comment Body",
                    },
                    "ref": null,
                    "rendered": "Comment Body",
                    "type": "label",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            "type": "form",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn white-text cyan darken-4 btn-flat"
            onClick={[Function]}
>
            Edit Comment
</button>,
          "className": "modal-footer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit Comment",
            "className": "btn white-text cyan darken-4 btn-flat",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Edit Comment",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modal-content"
>
            <h4>
                        Edit comment
            </h4>
            <form
                        className="col s12"
            >
                        <div
                                    className="row form-row"
                        >
                                    <div
                                                className="input-field col s12"
                                    >
                                                <textarea
                                                            className="materialize-textarea"
                                                            onChange={[Function]}
                                                            placeholder="Type some text"
                                                            type="text"
                                                            value="Edited comment body"
                                                />
                                                <label>
                                                            Comment Body
                                                </label>
                                    </div>
                        </div>
            </form>
</div>,
          <div
            className="modal-footer"
>
            <button
                        className="btn white-text cyan darken-4 btn-flat"
                        onClick={[Function]}
            >
                        Edit Comment
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Edit comment
</h4>,
              <form
                className="col s12"
>
                <div
                                className="row form-row"
                >
                                <div
                                                className="input-field col s12"
                                >
                                                <textarea
                                                                className="materialize-textarea"
                                                                onChange={[Function]}
                                                                placeholder="Type some text"
                                                                type="text"
                                                                value="Edited comment body"
                                                />
                                                <label>
                                                                Comment Body
                                                </label>
                                </div>
                </div>
</form>,
            ],
            "className": "modal-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit comment",
              },
              "ref": null,
              "rendered": "Edit comment",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="row form-row"
>
                  <div
                                    className="input-field col s12"
                  >
                                    <textarea
                                                      className="materialize-textarea"
                                                      onChange={[Function]}
                                                      placeholder="Type some text"
                                                      type="text"
                                                      value="Edited comment body"
                                    />
                                    <label>
                                                      Comment Body
                                    </label>
                  </div>
</div>,
                "className": "col s12",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="input-field col s12"
>
                    <textarea
                                        className="materialize-textarea"
                                        onChange={[Function]}
                                        placeholder="Type some text"
                                        type="text"
                                        value="Edited comment body"
                    />
                    <label>
                                        Comment Body
                    </label>
</div>,
                  "className": "row form-row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <textarea
                        className="materialize-textarea"
                        onChange={[Function]}
                        placeholder="Type some text"
                        type="text"
                        value="Edited comment body"
/>,
                      <label>
                        Comment Body
</label>,
                    ],
                    "className": "input-field col s12",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "materialize-textarea",
                        "onChange": [Function],
                        "placeholder": "Type some text",
                        "type": "text",
                        "value": "Edited comment body",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "textarea",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Comment Body",
                      },
                      "ref": null,
                      "rendered": "Comment Body",
                      "type": "label",
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
              "type": "form",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn white-text cyan darken-4 btn-flat"
              onClick={[Function]}
>
              Edit Comment
</button>,
            "className": "modal-footer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit Comment",
              "className": "btn white-text cyan darken-4 btn-flat",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Edit Comment",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
